### DISCUSSION TITLE
syntax error after splitting with pgn-extract

### DISCUSSION THREAD

**Discussion by arnaule** [rodpolako/Chess-PGN-Trainer#7] [General]
The main lines of my PGN is working with your software - it is perfect!

Trying to work the other lines i used pgn-extract as stated in your documentation, adding a -w1000 as i was having the 75 error.
It generated a file with about 60 games.

Now i am getting : SyntaxError: Expected "*","0-1","1-0","1/2"..... or whitespaces but end of input found.

For example the first game of the file generate the syntax error : 

[Event "?"]
[Site "?"]
[Date "2022.11.23"]
[Round "?"]
[White "?"]
[Black "?"]
[Result "*"]
[ECO "D31"]
[Annotator "?"]
[EventDate "2022.??.??"]
[PlyCount "36"]
[GameId "2076602978287251"]

{ [%evp 0,36,34,25,27,4,-11,-3,12,2,12,9,33,25,27,32,61,16,16,16,16,18,17,20,38,33,46,44,33,28,43,29,40,18,28,25,25,22,27] }

1. d4 d5 { [%mdl 32768] } 2. c4 e6 3. Nc3 a6 4. cxd5 exd5 { [%mdl 32768] } 5. Nf3 h6 6. Bf4 Nf6 7. e3 Bd6 8. Bxd6 Qxd6 9. Bd3 O-O 10. h3 Nc6 11. O-O Ne7 12. Qc2 g6 13. Na4 Bf5 14. Bxf5 Nxf5 15. Nc5 Rfb8 16. b4 c6 17. a4 Qc7 $1 18. Rab1 Nd6 *

Would you know what triggers the syntax error ?

**Comment by rodpolako** [comment]
The issue is with the section above the move order.  The parser expects the moves to follow after the last tag.  In your file you have an embedded command instead.  Remove the whole section starting with {[%evp as that isn't recognised by the parser (see the "Non-Standard PGNs" section under Known Limitations).  According to [this](https://chess.stackexchange.com/questions/26990/whats-the-meaning-of-evp-in-a-pgn-comment) %evp is a chessbase tag.   I took your pgn and removed the line and it now opens in the tool.  Notice that the moves follow after the last tag (GameID in this example):
```
[Event "?"]
[Site "?"]
[Date "2022.11.23"]
[Round "?"]
[White "?"]
[Black "?"]
[Result "*"]
[ECO "D31"]
[Annotator "?"]
[EventDate "2022.??.??"]
[PlyCount "36"]
[GameId "2076602978287251"]

1. d4 d5 { [%mdl 32768] } 2. c4 e6 3. Nc3 a6 4. cxd5 exd5 { [%mdl 32768] } 5. Nf3 h6 6. Bf4 Nf6 7. e3 Bd6 8. Bxd6 Qxd6 9. Bd3 O-O 10. h3 Nc6 11. O-O Ne7 12. Qc2 g6 13. Na4 Bf5 14. Bxf5 Nxf5 15. Nc5 Rfb8 16. b4 c6 17. a4 Qc7 $1 18. Rab1 Nd6 *
```

**Comment by arnaule** [comment]
thank you! it is strange because the same line was not causing an error from the original pgn not splitted.
I removed it from the orinigal pgn, and corrected another syntax error after the split (48 games out of 140 would load, because of a '--' before the last move) 
it works great now ! Such an amazing tool

**Comment by rodpolako** [comment]
Hi arnaule,
  You may want to try your original PGN with the variations again.  The application now supports variations directly so you no longer need to use pgn-extract to separate out the lines.  Also, I added logic to remove the double-hyphen notation and opened an issue with the author of the pgn-parser and we are disccussing that particular notation as it appears to be somewhat non-standard.

Even the original PGN in this issue now opens without issue in the new version.
